### PR TITLE
Map ViewModel

### PR DESCRIPTION
**Pull request by Polymeth**
This PR implements the viewmodel for the map. Now, the map is tied to the database and display markers/pins on task location depending on the chosen event.

Addresses #208 #125

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/ui/screens/event/tasktab/EventTaskViewModel.kt [hunk 1]
```diff
@@ -14,6 +15,7 @@ class EventTaskViewModel(val db: TaskAPI, val showSnackbar: (String) -> Unit) :
   init {
     uiState = _uiState
     updateTasks()
+    Log.w("mdrr: ", _uiState.value.tasks.size.toString())
```
Comment (ortoLover) [inline app/src/main/java/com/github/se/assocify/ui/screens/event/tasktab/EventTaskViewModel.kt:18]: You are leaving a debug message in the middle of the code.

File: app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  init {
+    // First, get all tasks
+    fetchTasks()
+    // Then, fetch the markers
+    fetchMarkers()
+  }
+
+  /** Fetch the tasks from the database */
+  fun fetchTasks() {
```
Comment (ortoLover) [inline app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt:25]: This function can be private

File: app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    // Set the loading state
+    _uiState.value = _uiState.value.copy(loading = true, error = null)
+    // Fetch the tasks from the database
+
+    taskAPI.getTasks(
+        { tasks ->
+          filterTasks(tasks)
+          _uiState.value = _uiState.value.copy(taskList = tasks, loading = false, error = null)
+        },
+        { _uiState.value = _uiState.value.copy(loading = false, error = "Error loading tasks") })
```
Comment (ortoLover) [inline app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt:35]: Check error as true here

**Review by ortoLover — CHANGES_REQUESTED** [review]
Good PR!! I have checked the final result on the app and it does what it is supposed to do, there is no more the problem with the screen moving. Also the code looks clean and expected a few nitpicks everything lookks good. Good job!!

File: app/src/main/java/com/github/se/assocify/ui/screens/event/EventScreenViewModel.kt [hunk 4]
```diff
@@ -101,16 +105,10 @@ class EventScreenViewModel(
...
-      EventPageIndex.Tasks -> {
-        taskListViewModel.search(_uiState.value.searchQuery)
-      }
-      EventPageIndex.Map -> {
-        /*TODO: implement for map screen*/
-      }
-      EventPageIndex.Schedule -> {
-        /*TODO: implement for schedule screen*/
-      }
+    if (_uiState.value.currentTab == EventPageIndex.Tasks) {
```
[the remark this answers is not in the document]
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/event/EventScreenViewModel.kt:108]: Was required for reliability, the CI didn't want the empty switch case even tho I never changed it lmao

File: app/src/main/java/com/github/se/assocify/ui/screens/event/tasktab/EventTaskViewModel.kt (hunk 1 above)
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/event/tasktab/EventTaskViewModel.kt:18]: Oupsie

File: app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt (hunk 3 above)
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/event/maptab/EventMapViewModel.kt:35]: No error so no needs to change error

**Review by ortoLover — COMMENTED** [review]
Everything good now!!

**Review by ortoLover — APPROVED** [review]
Forgot to approve, you can merge now!

**Pull request merged**